### DISCUSSION TITLE
docanalysis

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#12] [Announcements]
# docanalysis

https://github.com/petermr/petermr/docanalysis Text analysis and extraction from scientific articles (@shweatanhegde)

**Comment by ShweataNHegde** [comment]
I have been writing tests for potential use cases of the command line interface we hope to develop. 
 
https://github.com/petermr/docanalysis/blob/main/test/_test.py

**Comment by ShweataNHegde** [comment]
Based on discussion between AG and SNH on 2022-03-05

### TODO list (Higher to lower priority)
- Document
	- Command-line interface (SNH)
	- Caveats (SNH)
-  Section search
	- Incorporate muliple globs (SNH)
	- Let users to specify section of their interest (AG)
- Expand dictionary search.  (SNH/AG)
	- Match Synoymns 
	- Match Regex
	- Match Stemmed words
	- Fuzzy match
- Display In-line annotation after dictionary search (AG)
- Extract specific entities only
	- Let users specify the kind of Named Entity they want to extract (AG)

**Comment by petermr** [comment]
On Mon, Mar 7, 2022 at 11:33 AM Shweata N. Hegde ***@***.***>
wrote:

I think this looks good.

> TODO list (Higher to lower priority)
>
>    - Document
>       - Command-line interface (SNH)
>
> This will determine much of the rest. It's a statement of what we would
like the system to do as well as what it does

>
>    -
>       - Caveats (SNH)
>    - Section search
>       - Incorporate muliple globs (SNH)
>       - Let users to specify section of their interest (AG)
>
> This will be  important. Are they not effectively the same?

>
>    -
>       - Expand dictionary search. (SNH/AG)
>
> I think this is a good priority. Most of these should be standard calls in
NLTK or more basic Python

>
>    -
>       - Match Synoymns
>       - Match Regex
>       - Match Stemmed words
>       - Fuzzy match
>    - Display In-line annotation after dictionary search (AG)
>
> Presumably this comes from spaCy?

>
>    - Extract specific entities only
>       - Let users specify the kind of Named Entity they want to extract
>       (AG)
>
> These types will be determined by what spaCy has to offer?

-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by ShweataNHegde** [comment]
UPDATES (2022-03-09):
- Improved documentation (https://github.com/petermr/docanalysis#readme)
- Came up with a custom list of globs for sections (https://github.com/petermr/docanalysis/blob/main/docanalysis/glob_trail.py)
- Addressed comments on the manuscript

**Comment by ShweataNHegde** [comment]
2022-03-14 (PMR, AG, SNH)
- Search for synonyms
- Bag of words (at CProject (body), CTree)
- Move out the constants to an external file (globs, dictionaries)
- Display (entity in context, add links if spacy) 
- Histograms, correlations, etc.

**Comment by petermr** [comment]
# n-gram frequency

How far did you/we get with unsupervised n-grams? We'd like to extract meaningful phrases with counts from IPCC text.

**Reply by ShweataNHegde** [reply]
We currently have Named Entity Recognization (check this out: https://github.com/petermr/docanalysis/wiki/docanalysis-Tutorial#create-ami-dict-of-entities). I haven't added key phrase extraction functionality to `docanalysis`, but you can find some initial explorations, here: https://github.com/ShweataNHegde/snowball/blob/main/snowball_opendiagram_3.ipynb
